iOS-Email-Client/Controllers/EmailDetailViewController.swift

```diff
@@ -717,7 +720,8 @@ extension EmailDetailViewController: DetailMoreOptionsViewDelegate {
         let subject = "\(email.subject.lowercased().starts(with: "fw:") || email.subject.lowercased().starts(with: "fwd:") ? "" : "Fw: ")\(email.subject)"
         let image = UIImage(named: "footer_beta")
         let imageData:Data =  UIImagePNGRepresentation(image!)!
-        let html = Constants.singleEmail(image: imageData.base64EncodedString(), subject: subject, displayName: email.fromContact.displayName, email: email.fromContact.email, completeDate: email.completeDate, contacts: email.getFullContacts(), content: email.content)
+        var contact = email.from.isEmpty ? "\(email.fromContact.displayName) &lt;\(email.fromContact.email)&gt;" : email.from
```

Comment:
Better check If there is only email in `from` field. If that happens, get the name from the contact.